benchmark/worker/benchmark_client.go

```diff
@@ -273,28 +273,21 @@ func (bc *benchmarkClient) unaryLoop(conns []*grpc.ClientConn, rpcCountPerConn i
 				// The worker client needs to wait for some time after client is created,
 				// before starting benchmark.
 				if poissonLambda == nil { // Closed loop.
-					done := make(chan bool)
 					for {
-						go func() {
-							start := time.Now()
-							if err := benchmark.DoUnaryCall(client, reqSize, respSize); err != nil {
-								select {
-								case <-bc.stop:
-								case done <- false:
-								}
-								return
-							}
-							elapse := time.Since(start)
-							bc.lockingHistograms[idx].add(int64(elapse))
+						start := time.Now()
+						if err := benchmark.DoUnaryCall(client, reqSize, respSize); err != nil {
 							select {
 							case <-bc.stop:
```

Comment:
Sigh... it would have been so much nicer if this method accepted a context, which when cancelled would automatically lead to any ongoing RPCs getting cancelled and we can use that as a signal to return from this method as well. I guess cleanup work for another day.

Reply:
Do you want me to file an issue for this?

Reply:
Sure. TVC for the win!